apps/client/src/features/app-settings/panel/general-panel/StyleEditorModal.tsx

```diff
@@ -0,0 +1,109 @@
+import { lazy, useEffect, useRef, useState } from 'react';
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from '@chakra-ui/react';
+
+import { getCSSContents, postCSSContents, restoreCSSContents } from '../../../../common/api/assets';
+
+import style from './StyleEditorModal.module.scss';
+
+const CodeEditor = lazy(() => import('./StyleEditor'));
+
+interface CodeEditorModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export default function CodeEditorModal(props: CodeEditorModalProps) {
+  const { isOpen, onClose } = props;
+
+  const [css, setCSS] = useState('');
+  const [saveLoading, setSaveLoading] = useState(false);
+  const [resetLoading, setResetLoading] = useState(false);
+
+  const cssRef = useRef<string>(css);
```

Comment:
question: I wonder if the CSS ref is misplaced

It feels strange for the parent to control the contents of the child this way. 
The UI should be a function of state.
When the reset function is called, the contents change, this state should trigger a re-render which updates the Editor without us explicitely doing so